meshtastic/interfaces/ble/lifecycle_service.py

```
@@ -0,0 +1,1701 @@
+                iface._closed = True
+                if was_closing:
+                    logger.debug(
+                        "BLEInterface.close called while another shutdown is in progress; continuing with cleanup"
+                    )
+                if iface._state_manager._current_state not in (
+                    ConnectionState.DISCONNECTED,
+                    ConnectionState.DISCONNECTING,
+                ):
+                    if not iface._state_manager._transition_to(
+                        ConnectionState.DISCONNECTING
+                    ):
```

Ruff (0.15.6)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)